AsyncDisplayKit/Details/ASDataController.mm

```diff
@@ -461,8 +463,10 @@ - (void)endUpdatesAnimated:(BOOL)animated completion:(void (^)(BOOL))completion
     }];
     [_pendingEditCommandBlocks removeAllObjects];
 
-    [_mainSerialQueue performBlockOnMainThread:^{
-      [_delegate dataController:self endUpdatesAnimated:animated completion:completion];
+    [_editingTransactionQueue addOperationWithBlock:^{
```

Comment:
I can wait here as well. we just need all the editingTransaction to schedule their operations on the _mainSerialQueue before we schedule this. I think that might be cleaner.